tiny_dnn/core/framework/tensor.h

```diff
@@ -264,83 +405,61 @@ class Tensor {
     }
 
     void *mutable_device_data() {
+        static_assert(!kConst, "Non-constant operation on constant Tensor");
         toDevice();
         data_dirty_ = true;
         return (*device_data_)();
     }
 #endif
 
     size_t size() const {
-        return host_data_.size();
+        return calcSize();
```

Comment:
I have defined it, but it's yet commented since requires changes in constructors. 
As for isView, we might want to check if Tensor is of size of TensorStorage, but that's all, I believe. So IMHO no need for member variable for that